fault-proof/src/proposer.rs

```diff
@@ -835,48 +835,98 @@ where
 
     /// Fetch game from the factory.
     ///
-    /// Drop game if the game type is invalid or the output root is not valid.
-    /// Drop game if the parent game does not exist.
+    /// Drop game if:
+    /// - The game type is not supported.
+    /// - The game type deos not respect the expected type when created.
+    /// - The output root claim is invalid.
+    /// - The parent game does not exist in cache if it should have one.
     async fn fetch_game(&self, index: U256) -> Result<()> {
+        let mut state = self.state.lock().await;
```

Comment:
I'm pretty sure it's fine to hold lock here since we fetch games sequentially and all other tasks await for `sync_games`.
But just wanted to cross check with you since the best practice is to "lock only when we need to touch state".